### PR TITLE
Implement Dirac distribution (updated PR)

### PR DESCRIPTION
**Pull request by devmotion**
This is an update of https://github.com/JuliaStats/Distributions.jl/pull/861, according to the discussion over there.

The main differences are:
- Merged master and fixed merge conflicts
- Extended documentation
- Removed `kurtosis` and `skewness`
- Added `logpdf`
- Implement `eltype` for type instead of instance
- Use `cis` in the implementation of `cf`

I took some inspiration from `bernoulli.jl`.

Two questions:
- Would you like me to bump the version?
- I guess it is the convention in Distributions to return Float64 (at least it was suggested in the previous PR) - it seems in cases where we only return `0` or `1` this causes promotions in code of users that could be avoided?

### CODE REVIEW

**Comment by devmotion** [comment]
Travis tests don't run due to negative credit balance.

**Comment by mschauer** [comment]
Tests passing now with the new setup

**Comment by devmotion** [comment]
I think this PR is ready for review :slightly_smiling_face:

**Comment by devmotion** [comment]
Gentle bump :slightly_smiling_face:

File: src/univariate/discrete/dirac.jl [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+Base.eltype(::Type{Dirac{T}}) where {T} = T
+
+insupport(d::Dirac, x::Real) = x == d.value
+minimum(d::Dirac) = d.value
+maximum(d::Dirac) = d.value
+
+#### Properties
+
+mean(d::Dirac) = d.value
+var(d::Dirac) = 0.0
```
Comment (matbesancon) [inline src/univariate/discrete/dirac.jl:34]:
```suggestion
var(d::Dirac{T}) where {T} = zero(T)
```

File: src/univariate/discrete/dirac.jl [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+mode(d::Dirac) = d.value
+
+entropy(d::Dirac) = 0.0
```
Comment (matbesancon) [inline src/univariate/discrete/dirac.jl:38]:
```suggestion
entropy(d::Dirac{T}) where {T} = zero(T)
```

**Comment by matbesancon** [comment]
few minor comments, good to merge when these are addressed

**Comment by matbesancon** [comment]
thanks a lot for the PR!

**Comment by devmotion** [comment]
Thanks! Is it possible to make a new release?

**Comment by matbesancon** [comment]
sure yes, doing it now

**Comment by devmotion** [comment]
Thank you!

**Pull request merged**